File: app/Global.java
import java.io.File;
import java.io.IOException;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.util.List;
import java.util.Map;

import com.avaje.ebean.Ebean;
import com.avaje.ebean.SqlUpdate;
import com.avaje.ebean.TxRunnable;

import play.Play;
import play.Logger;
import play.Application;
import play.GlobalSettings;

import org.yaml.snakeyaml.Yaml;
import org.yaml.snakeyaml.constructor.CustomClassLoaderConstructor;


public class Global extends GlobalSettings {
    public void onStart(Application app) {
        File folder = new File("app/fixtures");
        if (folder == null || !folder.canRead()) {
            System.out.println("Couldn't load fixtures");
            return;
        }

        for (File file : folder.listFiles()) {
            loadFromFile(file);
        }
    }

    public void loadFromFile(File file) {
        Yaml yaml = new Yaml(
            new CustomClassLoaderConstructor(Play.application().classloader())
        );
        FileInputStream fis = null;

        try {
            fis = new FileInputStream(file);
        } catch (FileNotFoundException fnfe) { return; }

        Map<String, List<Object>> tableMap = (Map<String, List<Object>>) yaml.load(fis);  //yaml must be in conf folder?

        try {
            fis.close();
        } catch (IOException ioe) {}

        System.out.println("Load order: " + tableMap.entrySet());

        SqlUpdate disable = Ebean.createSqlUpdate("SET REFERENTIAL_INTEGRITY FALSE;"),
                  enable = Ebean.createSqlUpdate("SET REFERENTIAL_INTEGRITY TRUE;");

        Ebean.execute(new TxRunnable() {
            public void run() {
                Ebean.execute(disable); // disable;

                for (Map.Entry<String, List<Object>> tableEntry : tableMap.entrySet()) {
                    // delete old entries
                    Class<?> klass = tableEntry.getValue().getClass();

                    save(
                        tableEntry.getKey(),
                        tableEntry.getValue()
                    );
                }

                Ebean.execute(enable); // re-enable
            }
        });
    }

    private void save(String key, List<?> value) {
        Class<?> klass = value.get(0).getClass();

        Logger.info("Deleting {}", klass);
        Ebean.delete(Ebean.find(klass).findList());

        // add new entries
        Logger.info("Loading {}", key);
        Ebean.save(value);
    }
}

File: app/Global.scala
import java.io.File
import java.util.List
import java.util.ArrayList
import java.io.FileInputStream

import scala.collection.JavaConversions._

import play.Play
import play.Logger
import play.Application
import play.GlobalSettings
import play.db.ebean._

import com.avaje.ebean.Ebean
import org.yaml.snakeyaml.Yaml
import org.yaml.snakeyaml.constructor.CustomClassLoaderConstructor

import org.joda.time.DateTime;
import play.data.format.Formatters;
import views.formdata.JodaDateFormatter;


class Global extends GlobalSettings {
    override
    def onStart(app: Application) {

        Formatters.register(classOf[DateTime], new JodaDateFormatter())

        // val users = models.UserModel.find.all
        // if (users.size != 0) throw new Error("Restart application to load new fixtures")

        val folder = new File("app/fixtures")

        if (folder == null || !folder.canRead()) {
            play.Logger.error("Couldn't load fixtures")

        } else {
            play.Logger.info("Loading fixtures")

            disable_integrity()
            load_models(folder)
            enable_integrity()
        }
    }

    def exec(s: String) {
        Ebean.execute(Ebean.createSqlUpdate(s))
    }

    def load_models(folder: File) = {
        val all_models = (
            folder
            .listFiles
            .map(getYaml)
            .map(thing => thing.toList.map(_.toList))
        )

        for (model_file <- all_models) {
            for (models <- model_file) {
                save(new ArrayList(models))
            }
        }
    }

    def integrity(set: String) = () => exec("SET REFERENTIAL_INTEGRITY " + set + ";")

    val disable_integrity = integrity("FALSE")
    val  enable_integrity = integrity("TRUE")

    def getYaml(file: File) : List[List[Any]] = {
        val yaml = new Yaml(
            new CustomClassLoaderConstructor(Play.application().classloader())
        )

        val fis = new FileInputStream(file)

        if (fis != null) {
            val tableMap = yaml.load(fis)  //yaml must be in conf folder?

            fis.close

            return tableMap.asInstanceOf[List[List[Any]]]
        } else {
            throw new Error();
        }
    }

    def save(value: List[Any]) {
        val klass = value.toList.get(0).getClass()

        Logger.info("Deleting {}", klass)
        Ebean.delete(Ebean.find(klass).findList)

        // add new entries
        Logger.info("Loading {}", klass)
        value.foreach(Ebean.save)
    }
}
